src/Module.zig

```diff
@@ -3514,6 +3528,560 @@ fn semaFile(mod: *Module, file: *File) SemaError!void {
     }
 }
 
+const LowerZon = struct {
```

Comment:
I suggest moving this into a separate `LowerZon.zig`.
Also, parts of this type are basically reimplementing `Sema.zig` stuff. I recommend taking and storing a `Sema` for this, since it fundamentally *is* semantic analysis, just of a different syntax form.